plugins/PostProcessingPlugin/scripts/DisplayRemainingTimeOnLCD.py

```diff
@@ -0,0 +1,86 @@
+
+from ..Script import Script
+
+import re
+import datetime
+
+
+class DisplayRemainingTimeOnLCD(Script):
+
+    def __init__(self):
+        super().__init__()
+
+
+    def getSettingDataString(self):
+        return """{
+            "name":"Disaplay Remaining Time on LCD",
+            "key":"DisplayRemainingTimeOnLCD",
+            "metadata": {},
+            "version": 2,
+            "settings":
+            {
+                "TurnOn":
+                {
+                    "label": "Enable",
+                    "description": "When enabled, It will write Time Left HHMMSS on the display",
+                    "type": "bool",
+                    "default_value": true
+                }
+            }
+        }"""
+
+    def execute(self, data):
+        if self.getSettingValueByKey("TurnOn"):
+            TotalTime = 0           # Var for total time
+            TotalTimeString = ""    # Var for the string we insert
+            for layer in data:
+                layer_index = data.index(layer)
+                lines = layer.split("\n")
+                for line in lines:
+                    if line.startswith(";TIME:"):
+                        # At this point, we have found a line in the GCODE with ";TIME:"
+                        # which is the indication of TotalTime. Looks like: ";TIME:1337", where
+                        # 1337 is the total print time in seconds.
+                        line_index = lines.index(line)      # We take a hold of that line
+                        minString = re.split(":", line)     # Then we split it, so we can get the number
+
+                        StringMedTal = "{}".format(minString[1])    # Here we insert that number from the
+                                                                    # list into a string.
+
+                        TotalTime = int(StringMedTal)               # Only to contert it to a int.
+
+                        m, s = divmod(TotalTime, 60)    # Math to calculate
```

Comment:
We already have a nice convenience class that handles this for you (Duration).

You can use it like:
```python3
duration = Duration(total_time)
total_time_string = duration.getDisplayString()
```

Reply:
I have tried this, but I could not get it to work. Since what is in the code now works, and should continue to work I think that's fine.